src/ucp/wireup/wireup.c

```diff
@@ -636,16 +637,39 @@ ucp_wireup_connect_lane(ucp_ep_h ep, const ucp_ep_params_t *params,
                         const ucp_unpacked_address_t *remote_address,
                         unsigned addr_index)
 {
-    ucp_worker_h worker          = ep->worker;
-    ucp_rsc_index_t rsc_index    = ucp_ep_get_rsc_index(ep, lane);
-    ucp_lane_index_t proxy_lane  = ucp_ep_get_proxy_lane(ep, lane);
-    ucp_worker_iface_t *wiface   = ucp_worker_iface(worker, rsc_index);
+    ucp_worker_h worker       = ep->worker;
+    unsigned cm_ep_init_flags = ucp_cm_ep_init_flags(worker, params);
+    int connect_aux;
+    ucp_lane_index_t proxy_lane;
+    ucp_rsc_index_t rsc_index;
+    ucp_worker_iface_t *wiface;
     uct_ep_params_t uct_ep_params;
     uct_ep_h uct_ep;
     ucs_status_t status;
 
     ucs_trace("ep %p: connect lane[%d]", ep, lane);
 
+    if ((lane == ucp_ep_get_cm_lane(ep)) &&
+        (cm_ep_init_flags & UCP_EP_INIT_CM_WIREUP_SERVER)) {
+        /* create a server side CM endpoint */
+        ucs_trace("ep %p: uct_ep[%d]", ep, lane);
+        return ucp_wireup_connect_cm_lane(ep, params);
+    }
+
+    if ((remote_address == NULL) || (remote_address->address_list == NULL) ||
```

Comment:
i think these checks should be assertions
it's not user/system error but indicates a bug